dist/cjs/trustedtypes.api_only.js

```diff
@@ -278,6 +434,91 @@ const trustedTypesBuilderTestOnly = function() {
     return exposedPolicies.get(pName) || null;
   }
 
+  /**
+   * Returns the name of the trusted type required for a given element
+   *   attribute.
+   * @param {string} tagName The name of the tag of the element.
+   * @param {string} attribute The name of the attribute.
+   * @param {string=} elementNs Element namespace.
+   * @param {string=} attributeNs The attribute namespace.
+   * @return {string|undefined} Required type name or undefined, if a Trusted
+   *   Type is not required.
+   */
+  function getAttributeType(tagName, attribute, elementNs = '',
+      attributeNs = '') {
+    // TODO: Support namespaces.
+    const canonicalAttr = toLowerCase.apply(String(attribute));
+    return getTypeInternal_(tagName, 'attributes', canonicalAttr);
+  }
+
+  /**
+   * Returns a type name from a type map.
+   * @param {string} tag A tag name.
+   * @param {string} container 'attributes' or 'properties'
+   * @param {string} property The attribute / property name.
+   * @return {string|undefined}
+   * @private
+   */
+  function getTypeInternal_(tag, container, property) {
+    if (HTML_TYPE_MAP['*'][container][property]) {
+      return HTML_TYPE_MAP['*'][container][property];
+    }
+
+    const canonicalTag = toUpperCase.apply(String(tag));
+
+    if (!HTML_TYPE_MAP[canonicalTag]) {
```

Comment:
Ought this be a hasOwnProperty check?

Or ought the copy pass above do Object.setPrototypeOf(x, null) for all objects where properties correspond to tags or property/attribute names?

Or just use Map instead of {}.

This also affects the [property] lookups above and below.